core/src/main/java/org/apache/accumulo/core/data/ResourceGroupId.java

```diff
@@ -18,15 +18,18 @@
  */
 package org.apache.accumulo.core.data;
 
+import java.util.regex.Pattern;
+
 import org.apache.accumulo.core.Constants;
-import org.apache.accumulo.core.conf.cluster.ClusterConfigParser;
 import org.apache.accumulo.core.util.cache.Caches;
 import org.apache.accumulo.core.util.cache.Caches.CacheName;
 
 import com.github.benmanes.caffeine.cache.Cache;
 
 public class ResourceGroupId extends AbstractId<ResourceGroupId> {
 
+  public static final Pattern GROUP_NAME_PATTERN = Pattern.compile("^[a-zA-Z]+(_?[a-zA-Z0-9])*$");
```

Comment:
This can be simplified substantially:
```suggestion
  public static final Pattern GROUP_NAME_PATTERN = Pattern.compile("^[a-zA-Z]\w*$");
```

The main difference with this is that you can have repeated or trailing underscores, which I think is probably fine, if users really want such weird names.

Reply:
IIRC I found the regex somewhere because I ran into the problem of the resource group name including characters that were invalid for a bash variable name. The `cluster.yaml` file is parsed and the contents written out to a file which is then sourced by `accumulo-cluster`. Happy to replace with a different regex, just need to make sure that it's compliant with bash variable names.

Reply:
My version still restricts it to characters that are valid bash variable names. Bash allows variables that start with underscores, end with them, or have repeated ones. So, both your version and mine are more restrictive than bash, in that it requires a leading letter character. Yours is stricter than mine, but I think mine is easier to understand.... however, I forgot that Java requires double escaping backslashes in String literals, so I think it should actually be:

```suggestion
  public static final Pattern GROUP_NAME_PATTERN = Pattern.compile("^[a-zA-Z]\\w*$");
```